crates/metadata/src/layout/mod.rs

```diff
@@ -241,6 +247,10 @@ where
     pub fn ty(&self) -> &F::Type {
         &self.ty
     }
+
+    pub fn new_from_ty(key: LayoutKey, ty: <F as Form>::Type) -> Self {
```

Comment:
I think this constructor should become `new` because it allows setting all fields. The existing `new<T>` can be renamed to `from_key<T>` or similar to be consistent.